### ISSUE TITLE
Cross-compiled gRPC is missing binaries for the target

### ISSUE THREAD

**Issue by xhochy** [grpc/grpc#26857]
This is a follow-up issue to the PR https://github.com/grpc/grpc/pull/26148 that introduced the problem. I could just send a PR that reverts it but that reintroduces the problem again that led to that PR. I sadly don't know the intention of their PR as we have been cross-compiling gRPC successfully since at least 1.32.

### What version of gRPC and what language are you using?

1.39

### What operating system (Linux, Windows,...) and version?

Linux and macOS

### What runtime / compiler are you using (e.g. python version or version of gcc)

Various; this issue is independent of that.

### What did you do?

We are building gRPC through cross-compilation for e.g. Apple M1 or Linux aarch64 and providing the output as conda packages that are intended to be used on these architectures. There you will need a native version of `grpc_cpp_plugin` to natively build packages that depend on gRPC.

### What did you expect to see?

`$PREFIX/bin/grpc_cpp_plugin` should be built and installed into the target prefix.

### What did you see instead?

The above-mentioned plugin was neither built nor installed. This makes sense if the build of gRPC would only be used in a cross-compiling environment but the output of the mentioned build here will be used for cross-compiling as well as compiling natively on the target system later.

### Anything else we should know about your project / environment?

We will revert the PR https://github.com/grpc/grpc/pull/26148 in conda-forge for now for the 1.39 builds to get the users of the cross-compiled package back to a working state.

**Comment by veblush** [comment]
@xhochy Our of curiosity, why do you use `install` for building packages? I guess it might be for getting built artifacts. Is there an alternative way for getting those?

@UVV-gh Could you explain how installing cross-compiled protoc was a problem for your case?

**Comment by UVV-gh** [comment]
It seems like @xhochy has a different use case, he actually uses cross-compiled gRPC plugins on the target. Yocto-based generated SDK is used on the _host_ for cross-compilation for the target. SDK consists of two sysroots, one for the target and one for the host, the latter would contain gRPC plugins, which would have to be searched manually and substituted to CMake.

**Comment by clementperon** [comment]
There is a pull request to have a proper cross compilation support between gRPC libs and gRPC binaries: 
https://github.com/grpc/grpc/pull/22498

I add it to my Yocto and I have now a proper SDK and a proper cross compilation.

**Comment by xhochy** [comment]
> @xhochy Our of curiosity, why do you use `install` for building packages? I guess it might be for getting built artifacts. Is there an alternative way for getting those?

I'm a bit confused about this question. For CMake-based workflows `cmake … && make install` is the normal way to build and install artefacts. Can you maybe rephrase this?

**Comment by xhochy** [comment]
@UVV-gh would it work in your case if there was a special flag that disabled the installation of the plugins instead of fully removing it from the CMake setup? I would expect the default output of a cross-compiled run to be equal to the non-cross compiled one. The consumers (on the target) of gRPC should in the end not be able to differentiate them.

**Comment by UVV-gh** [comment]
@xhochy yes, I'm open to any solution, for example something like `gRPC_BUILD_CROSS_SDK` would work for me
